packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts

```diff
@@ -251,3 +251,128 @@ export const estimatePendingRewards = async ({ api,
+  {
+    api, extrinsicHash, height
+  }:
+    {
+      api: ApiPromise, extrinsicHash: string, height: number
+    }): Promise<{ claimedRewards: number }> => {
+  try {
+
+    const blockHash = await api.rpc.chain.getBlockHash(height);
+    const signedBlock = await api.rpc.chain.getBlock(blockHash);
+    const apiAt = await api.at(signedBlock.block.header.hash);
+    const allRecords = (await apiAt.query.system.events()).toArray();
+
+    // Find the extrinsic index by matching the extrinsic hash
+    const extrinsicIndex = signedBlock.block.extrinsics.findIndex(
+      (ex) => ex.hash.toString() === extrinsicHash
+    );
+
+    if (extrinsicIndex === -1) {
+      throw new Error('Extrinsic not found in the block');
+    }
+
+    // Get events associated with the extrinsic
+    const extrinsicEvents = allRecords.filter((record) =>
+      record.phase.isApplyExtrinsic &&
+      record.phase.asApplyExtrinsic.eq(extrinsicIndex)
+    );
+
+    const methodRwd = 'Reward';
+    const methodBns = 'BonusReward';
+    const section = 'dappStaking';
+    let claimedReward = BigInt('0');
+
+    extrinsicEvents.map((e, idx) => {
+      console.log(e.toHuman(), idx);
+      if ((e.toHuman()?.event?.method == methodRwd || e.toHuman()?.event?.method == methodBns) &&
+        e.toHuman()?.event?.section == section) {
+        let tmpAmount = e.toHuman().event?.data?.amount;
+        let tmpAmountBigInt = commaStrToBigInt(tmpAmount);
+
+        claimedReward += tmpAmountBigInt;
+
+      } else {
+        claimedReward = claimedReward;
+      }
+    })
+
+    return { claimedRewards: Number(claimedReward) };
+
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+}
+
+/**
+ * Memo:
+ * This method returns usedfee amount by extrinsicHash and blockHeight that have the actualFee.
+ */
+export const UsedFee = async (
+  {
+    api, extrinsicHash, height
+  }:
+    {
+      api: ApiPromise, extrinsicHash: string, height: number
+    }): Promise<{ usedFee: number }> => {
+  try {
+
+    const blockHash = await api.rpc.chain.getBlockHash(height);
+    const signedBlock = await api.rpc.chain.getBlock(blockHash);
+    const apiAt = await api.at(signedBlock.block.header.hash);
+    const allRecords = (await apiAt.query.system.events()).toArray();
+
+    // Find the extrinsic index by matching the extrinsic hash
+    const extrinsicIndex = signedBlock.block.extrinsics.findIndex(
+      (ex) => ex.hash.toString() === extrinsicHash
+    );
+
+    if (extrinsicIndex === -1) {
+      throw new Error('Extrinsic not found in the block');
+    }
+
+    // Get events associated with the extrinsic
+    const extrinsicEvents = allRecords.filter((record) =>
+      record.phase.isApplyExtrinsic &&
+      record.phase.asApplyExtrinsic.eq(extrinsicIndex)
+    );
+
+    const method = 'TransactionFeePaid';
+    const section = 'transactionPayment';
+
+    let usedFee = BigInt('0');
+
+    extrinsicEvents.map((e) => {
+      if (e.toHuman()?.event?.method == method &&
```

Comment:
Can you avoid using `toHuman` and use polkadot.js types instead. This will also remove need for `commaStrToBigInt`
The same applies for the method above